Advanced Solo Setup Docs — Analysis

## Critical Errors

### 1. Three Falcon component toggle flags do not exist — `falcon-deployment.md`

**Location:** [`falcon-deployment.md`](content/en/docs/advanced-solo-setup/network-deployments/falcon-deployment.md) — "Component Toggles" section

The docs describe three flags for controlling which optional components are deployed:

```bash
solo one-shot falcon deploy \
  --values-file falcon-values.yaml \
  --deploy-explorer=false \
  --deploy-relay=false
```

With a reference table:

| Flag | Default | Description |
|---|---|---|
| `--deploy-mirror-node` | `true` | Include the mirror node |
| `--deploy-explorer` | `true` | Include the explorer |
| `--deploy-relay` | `true` | Include the JSON RPC relay |

None of these three flags exist in the codebase.

**Source:** `src/commands/one-shot/default-one-shot.ts:71-73` — the complete `FALCON_DEPLOY_FLAGS_LIST` is:

```typescript
public static readonly FALCON_DEPLOY_FLAGS_LIST: CommandFlags = {
  required: [],
  optional: [flags.quiet, flags.valuesFile, flags.numberOfConsensusNodes],
};
```

There is no `deployMirrorNode`, `deployExplorer`, `deployRelay`, `deploy-mirror-node`, `deploy-explorer`, or `deploy-relay` anywhere in `src/commands/flags.ts` or `src/commands/one-shot/`. An exhaustive search found only `deployJsonRpcRelay` (`--json-rpc-relay`, `src/commands/flags.ts:513`), which appears in a different context.

**Impact:** Users following the "Component Toggles" section to skip the explorer or relay will get "Unknown argument" errors. This section documents a feature that does not exist, which is especially misleading because it presents a clean, well-formatted reference table as authoritative.

**Suggested fix:** Remove or clearly mark the entire "Component Toggles" section as not yet implemented. If this is a planned feature, add a note such as "Component toggles are not yet available. To deploy without optional components, omit their sections from the values file — Solo skips any section not present in the file."

---

### 2. `--no-rollback` flag does not exist — `falcon-deployment.md`

**Location:** [`falcon-deployment.md`](content/en/docs/advanced-solo-setup/network-deployments/falcon-deployment.md) — "Rollback and Failure Behaviour" section

The docs describe automatic rollback on failure and offer a flag to disable it:

```bash
solo one-shot falcon deploy \
  --values-file falcon-values.yaml \
  --no-rollback
```

The flag `--no-rollback` does not exist. It is not in `FALCON_DEPLOY_FLAGS_LIST` (`src/commands/one-shot/default-one-shot.ts:71-73`), not in `src/commands/flags.ts`, and not referenced anywhere in the one-shot deploy logic.

**Source:** The `FALCON_DEPLOY_FLAGS_LIST` only contains: `quiet`, `valuesFile`, `numberOfConsensusNodes`. No rollback-related flag is present in any related source file.

**Impact:** Running `solo one-shot falcon deploy --values-file falcon-values.yaml --no-rollback` will produce an "Unknown argument: no-rollback" error. The description of automatic rollback behavior itself may also be inaccurate — the deploy logic does not appear to implement automatic rollback on failure (no rollback logic was found in `deployInternal` in `default-one-shot.ts`).

**Suggested fix:** Remove the `--no-rollback` flag reference. If rollback behavior exists in some form (e.g., via the standard `--force` flag or no rollback at all), document the actual behavior. If this is a planned feature, note it as such.

---

### 3. `solo one-shot show deployment` and `solo deployment refresh port-forwards` do not exist — `falcon-deployment.md`

**Location:** [`falcon-deployment.md`](content/en/docs/advanced-solo-setup/network-deployments/falcon-deployment.md) — "Deployment Output" section

The docs document two commands for post-deployment inspection:

```bash
solo one-shot show deployment
```

```bash
solo deployment refresh port-forwards
```

Neither command exists.

**Source — `solo one-shot show deployment`:** `src/commands/command-definitions/one-shot-command-definition.ts` defines exactly six subcommands: `single deploy`, `single destroy`, `multi deploy`, `multi destroy`, `falcon deploy`, `falcon destroy`. There is no `show` group or `show deployment` command.

**Source — `solo deployment refresh port-forwards`:** `src/commands/command-definitions/deployment-command-definition.ts` defines only: `cluster attach`, `config list`, `config create`, `config delete`, `diagnostics all`, `diagnostics logs`, `diagnostics connections`. There is no `refresh` subcommand. Port-forward refresh is not a stand-alone CLI command — port forwarding is re-established when component commands are re-run (e.g., `solo consensus node start`).

**Impact:** Both commands will produce "command not found" errors. The "Deployment Output" section appears to be documenting a planned UI that has not been implemented yet.

**Suggested fix:**
- Remove `solo one-shot show deployment`. To inspect deployment output, direct users to the `~/.solo/one-shot-<deployment>/` directory directly.
- Remove `solo deployment refresh port-forwards`. To restore interrupted port-forwards, document the actual mechanism (rerunning `solo consensus node start`, `solo mirror node add`, etc. with the component already installed will re-enable port forwarding due to the `reuse` flag in `managePortForward`).

---

### 4. Wrong npm package name in CI workflow — `solo-ci-workflow.md`

**Location:** [`solo-ci-workflow.md`](content/en/docs/advanced-solo-setup/solo-ci-workflow.md) — Step 3 and Complete Example Workflow

The CI workflow installs Solo with:

```yaml
npm install -g @hiero-ledger/solo@0.68.0
```

The correct npm package name is `@hashgraph/solo`.

**Source:** `package.json:2`:

```json
"name": "@hashgraph/solo",
```

`@hiero-ledger/solo` does not appear in `package.json`, `package-lock.json`, or any build/publish configuration in the Solo repository. Installing `@hiero-ledger/solo` will fail with "npm ERR! 404 Not Found".

**Impact:** Any CI pipeline copied from this guide will fail at the npm install step. This is the most immediately actionable bug in the entire Advanced Solo Setup section — it would affect every user trying to integrate Solo into CI.

**Suggested fix:**

```yaml
npm install -g @hashgraph/solo@<version>
solo --version
```

The pinned version `0.68.0` is also inconsistent with the compatibility table in system-readiness.md (which shows max version `0.59.x`). Use the current latest stable version or make the version a workflow input variable.

---

### 5. `submit-transaction` should be `submit-transactions` — `consensus-node-operations.md`

**Location:** [`consensus-node-operations.md`](content/en/docs/advanced-solo-setup/network-deployments/consensus-node-operations.md) — all three node operation workflows (add, update, delete)

Every workflow uses `submit-transaction` (singular):

```bash
solo consensus dev-node-add submit-transaction --deployment ...
solo consensus dev-node-update submit-transaction --deployment ...
solo consensus dev-node-delete submit-transaction --deployment ...
```

The correct subcommand name is `submit-transactions` (plural).

**Source:** `src/commands/command-definitions/consensus-command-definition.ts:62`:

```typescript
public static readonly DEV_NODE_SUBMIT_TRANSACTION = 'submit-transactions';
```

This constant is used for all three dev-node command groups (add at line 267, update at line 301, delete at line 365).

**Impact:** Running any of the three `submit-transaction` variants will produce a "command not found" error. This affects every node lifecycle operation on this page.

**Suggested fix:** Replace all three occurrences of `submit-transaction` with `submit-transactions`.

---

## Moderate Issues

### 6. Manual deployment prerequisite references `solo init` — `manual-deployment.md`

**Location:** [`manual-deployment.md`](content/en/docs/advanced-solo-setup/network-deployments/manual-deployment.md) — "Prerequisites" section, line 29

The prerequisites include:

> **Quickstart** — you have a running Kind cluster and have run `solo init` at least once.

The `solo init` command exists in the source (`src/commands/init/init.ts`) and is registered as a top-level command. However:

1. The manual deployment steps do not call `solo init` anywhere in the 10-step process.
2. The Quickstart guide (`content/en/docs/simple-solo-setup/quickstart.md`) does not mention `solo init` — it only runs `solo one-shot single deploy`.
3. In the Simple Solo Setup analysis, it was established that `solo init` is the deprecated predecessor to the one-shot flow and is no longer part of the recommended path.

**Impact:** A user following the Quickstart (as directed) will not have run `solo init`. If `solo init` is genuinely required before manual deployment, this prerequisite needs to explain why and what it does. If it is not required, the prerequisite should be removed to avoid confusion.

**Suggested fix:** Determine whether `solo init` is actually required before manual deployment. If not, remove the reference. If it is, add a step: `solo init` with its expected output and a brief explanation of what it installs (kubectl, helm, etc.). This is particularly important because `solo init` behavior is what installs kubectl and Helm — which has implications for the system-readiness auto-install claims (documented in the Simple Solo Setup analysis, issue #4).

---

### 7. CI workflow uses Kind v0.26.0, below the enforced minimum — `solo-ci-workflow.md`

**Location:** [`solo-ci-workflow.md`](content/en/docs/advanced-solo-setup/solo-ci-workflow.md) — Step 1 (Set Up Kind)

The CI workflow installs Kind using:

```yaml
uses: helm/kind-action@...
with:
  version: v0.26.0
```

The Solo source enforces a minimum Kind version of `v0.29.0` at runtime (`version.ts:17`; `src/integration/kind/impl/default-kind-client.ts:72-76`). Installing Kind `v0.26.0` will cause `solo one-shot single deploy` to fail immediately with a `KindVersionRequirementException`.

**Suggested fix:** Update to `version: v0.29.0` (or later). Also align the `kubectl_version` and `node_image` values, which currently reference `v1.31.4` — verify this Kubernetes version is still compatible with the current Solo Kind node image.

---

### 8. Component version defaults in env vars doc are stale — `using-environment-variables.md`

**Location:** [`using-environment-variables.md`](content/en/docs/advanced-solo-setup/using-environment-variables.md) — "Component Versions" section

The documented default versions differ from the actual defaults in `version.ts`:

| Variable | Docs say | Source (`version.ts`) says |
|---|---|---|
| `CONSENSUS_NODE_VERSION` | `v0.65.1` | `v0.68.6` |
| `BLOCK_NODE_VERSION` | `v0.18.0` | `v0.23.2` |
| `MIRROR_NODE_VERSION` | `v0.138.0` | `v0.145.2` |
| `EXPLORER_VERSION` | `v25.1.1` | `25.1.1` (no `v` prefix) |
| `RELAY_VERSION` | `v0.70.0` | `0.73.0` (no `v` prefix) |
| `SOLO_CHART_VERSION` | `v0.56.0` | `0.59.0` (no `v` prefix) |

**Source:** `C:\repos\solo\version.ts` lines 22–29.

**Impact:** Users relying on this table to set environment variables for version pinning will pin to outdated versions. The stale versions also create a consistency issue with the `solo --version` and version compatibility table in system-readiness.md.

How will we keep this updated moving forward? Should we remove?
---

### 9. `DEFAULT_START_ID_NUMBER` default is documented with account format, source uses raw integer — `using-environment-variables.md`

**Location:** [`using-environment-variables.md`](content/en/docs/advanced-solo-setup/using-environment-variables.md) — "Network and Node Identity" section

The docs show:

| Variable | Default |
|---|---|
| `DEFAULT_START_ID_NUMBER` | `0.0.3` |

**Source:** `src/core/constants.ts:210`:

```typescript
export const DEFAULT_START_ID_NUMBER: number = +getEnvironmentVariable('DEFAULT_START_ID_NUMBER') || 3;
```

The constant is a raw integer (`3`), not a Hedera account ID string (`0.0.3`). The `0.0.3` format is the full account ID representation that Solo builds from this number — it is not what you set the environment variable to.

**Impact:** A user who sets `DEFAULT_START_ID_NUMBER=0.0.3` is setting the value to `NaN` (since `+("0.0.3")` is `NaN`), which would fall through to the default `3` — effectively a no-op rather than an error. But a user who reads this as "this is the account ID format" may misunderstand the variable's scope.

**Suggested fix:** Change the documented default to `3` and add a description note: "Raw node ID number. The first consensus node account is `0.0.<DEFAULT_START_ID_NUMBER>`."

---

### 10. `NODE_CLIENT_PING_INTERVAL` is documented but does not exist — `using-environment-variables.md`

**Location:** [`using-environment-variables.md`](content/en/docs/advanced-solo-setup/using-environment-variables.md) — "Node Client Behaviour" section

The docs list:

| Variable | Default |
|---|---|
| `NODE_CLIENT_PING_INTERVAL` | `30000` |

This variable does not exist in `src/core/constants.ts`. An exhaustive search for `PING_INTERVAL`, `pingInterval`, and `ping_interval` returned no matches in constants.ts. The related variables that do exist are `NODE_CLIENT_SDK_PING_MAX_RETRIES` and `NODE_CLIENT_SDK_PING_RETRY_INTERVAL`.

**Impact:** Setting `NODE_CLIENT_PING_INTERVAL` has no effect — Solo never reads it. Users relying on it to tune ping behavior are configuring a variable Solo ignores.

**Suggested fix:** Remove `NODE_CLIENT_PING_INTERVAL` from the table. If there is an interval that was recently renamed or removed, note the migration path.

---

### 11. Two environment variables are missing from the reference — `using-environment-variables.md`

**Location:** [`using-environment-variables.md`](content/en/docs/advanced-solo-setup/using-environment-variables.md) — "Block Node" section

The following variables exist in `src/core/constants.ts` (lines 387–389) but are not documented:

| Variable | Default | Controls |
|---|---|---|
| `BLOCK_NODE_PODS_RUNNING_MAX_ATTEMPTS` | `900` | Maximum attempts to check if block node pods are running |
| `BLOCK_NODE_PODS_RUNNING_DELAY` | `1000` | Interval between block node pod running checks (ms) |

The Block Node section currently documents only the active/timeout checks, not the running checks. This is inconsistent with the Relay Node section which documents both `RELAY_PODS_RUNNING_MAX_ATTEMPTS` and `RELAY_PODS_RUNNING_DELAY`.

**Suggested fix:** Add both variables to the Block Node section, following the same pattern as the Relay Node section.

---

## Minor Issues

### 12. `SOLO_CHAIN_ID` is listed twice in the env vars doc — `using-environment-variables.md`

**Location:** [`using-environment-variables.md`](content/en/docs/advanced-solo-setup/using-environment-variables.md)

`SOLO_CHAIN_ID` appears in both:
- "General" section (line 31): `Chain ID of the Solo network`, default `298`
- "Network and Node Identity" section (line 46): `Chain ID of the Solo network`, default `298`

The variable exists once in source (`src/core/constants.ts`). One occurrence should be removed.

---

### 13. Broken table formatting — `customizing-solo-with-tasks.md`

**Location:** [`customizing-solo-with-tasks.md`](content/en/docs/advanced-solo-setup/customizing-solo-with-tasks.md) — "Infrastructure Tasks" table, line 290

The markdown table for Infrastructure Tasks has a raw `\n` escape sequence embedded inside the table header separator row:

```markdown
| Task                        | Description                                                |\n| --------------------------- | ---------------------------------------------------------- |
```

This will render as literal text `\n` rather than a line break in the markdown table. This is a copy-paste artifact from an escaped string.

---

### 14. Manual deployment cleanup is missing the Explorer destroy step — `manual-deployment.md`

**Location:** [`manual-deployment.md`](content/en/docs/advanced-solo-setup/network-deployments/manual-deployment.md) — "Cleanup" section

The cleanup section documents four destroy steps:
1. Destroy JSON-RPC Relay
2. Destroy Mirror Node
3. Destroy Explorer
4. Destroy the Network

The deployment in steps 1–10 deploys Explorer (step 9), but the cleanup section lists "Destroy Mirror Node" before "Destroy Explorer" (steps 2 and 3). The comment at the top says "destroy components in the reverse order of deployment." The deployment order is: relay (step 10) → explorer (step 9) → mirror (step 8), so the correct reverse destroy order should be: relay → explorer → mirror → network.

The current cleanup order (relay → mirror → explorer → network) inverts mirror and explorer. While this may not cause a hard failure (both are independent deployments), it contradicts the stated principle and could matter if there are ingress controller dependencies between explorer and mirror.

---

### 15. Falcon defaults tip states wrong cluster-ref default — `falcon-deployment.md`

**Location:** [`falcon-deployment.md`](content/en/docs/advanced-solo-setup/network-deployments/falcon-deployment.md) — "What Falcon Does Not Read from the File" section

The docs state:

> **Tip:** When not specified, Falcon uses these defaults: `--deployment one-shot`, `--namespace one-shot`, `--cluster-ref one-shot`, and `--num-consensus-nodes 1`.

For comparison, the standard `solo one-shot single deploy` generates a unique deployment name (`solo-<uuid>`), not a fixed `one-shot`. This is only the default for the Falcon path, not the one-shot path. If this is specific to the Falcon subcommand, the tip should explicitly say "Falcon defaults" rather than implying this is a universal one-shot default. This may also be worth verifying against the actual `deployFalcon` initialization logic — the docs source confirms the values `one-shot` as the Falcon-specific defaults, but users may be confused by the inconsistency with the non-Falcon one-shot defaults.

---

### 16. Manual deployment expected output shows Solo version 0.63.0, inconsistent with compatibility table — `manual-deployment.md`

**Location:** [`manual-deployment.md`](content/en/docs/advanced-solo-setup/network-deployments/manual-deployment.md) — expected output blocks throughout

All expected output blocks show `Version : 0.63.0`. The compatibility table in system-readiness.md lists `0.59.0` as the latest (non-LTS) release. This is a stale output example that will confuse users whose actual output shows a different version.

**Suggested fix:** Either update the version in all output blocks to match the current release, or add a note above the first output block: "Version numbers in example output may differ from your installation."

---

## Summary — Environment Variables Accuracy

| Variable | Status | Notes |
|---|---|---|
| `SOLO_HOME`, `SOLO_CACHE_DIR`, `SOLO_LOG_LEVEL`, `SOLO_DEV_OUTPUT`, `SOLO_CHAIN_ID`, `FORCE_PODMAN` | ✓ Correct | All confirmed in constants.ts |
| `DEFAULT_START_ID_NUMBER` | ✗ Wrong default format | Documented as `0.0.3`, source is raw integer `3` |
| `SOLO_NODE_INTERNAL/EXTERNAL_GOSSIP_PORT`, `SOLO_NODE_DEFAULT_STAKE_AMOUNT`, `GRPC_PORT`, `LOCAL_NODE_START_PORT` | ✓ Correct | Confirmed |
| `SOLO_OPERATOR_ID`, `SOLO_OPERATOR_KEY`, `SOLO_OPERATOR_PUBLIC_KEY`, `FREEZE_ADMIN_ACCOUNT`, `GENESIS_KEY` | ✓ Correct | Confirmed |
| `NODE_CLIENT_PING_INTERVAL` | ✗ Does not exist | Remove from docs |
| All other Node Client Behaviour variables | ✓ Correct | Confirmed |
| All Pod and Network Readiness variables | ✓ Correct | Confirmed |
| `BLOCK_NODE_PODS_RUNNING_MAX_ATTEMPTS`, `BLOCK_NODE_PODS_RUNNING_DELAY` | ✗ Missing | Add to Block Node section |
| All other Block Node variables | ✓ Correct | Confirmed |
| All Relay Node variables | ✓ Correct | Confirmed |
| All Load Balancer, Lease Management variables | ✓ Correct | Confirmed |
| All component version defaults | ✗ Stale | See issue #8 for current values |
| All Helm Chart URLs | ✓ Correct | Confirmed against constants.ts |
| All Network Load Generator variables | ✓ Correct | Confirmed |
| `ONE_SHOT_WITH_BLOCK_NODE`, `MIRROR_NODE_PINGER_TPS` | ✓ Correct | Confirmed |
| `SOLO_CHAIN_ID` (duplicated) | ✗ Duplicate entry | Remove from Network and Node Identity section |

--- a/content/en/docs/advanced-solo-setup/customizing-solo-with-tasks.md
+++ b/content/en/docs/advanced-solo-setup/customizing-solo-with-tasks.md
@@ -288,7 +288,8 @@ task solo:block:add
 
 Low-level tasks for managing clusters and network infrastructure:
 
-| Task                        | Description                                                |\n| --------------------------- | ---------------------------------------------------------- |
+| Task                        | Description                                                |
+| --------------------------- | ---------------------------------------------------------- |
 | `cluster:create`            | Create a Kind (Kubernetes in Docker) cluster               |
 | `cluster:destroy`           | Delete the Kind cluster                                    |
 | `solo:cluster:setup`        | Setup cluster infrastructure and prerequisites             |

--- a/content/en/docs/advanced-solo-setup/network-deployments/falcon-deployment.md
+++ b/content/en/docs/advanced-solo-setup/network-deployments/falcon-deployment.md
@@ -16,9 +16,7 @@ One-shot Falcon deployment is Solo's YAML-driven one-shot workflow. It uses the 
 deployment pipeline as `solo one-shot single deploy`, but lets you inject
 component-specific flags through a single values file.
 
-One-shot use Falcon deployment when you need a repeatable advanced setup, want to check a
-complete deployment into source control, or need to customise component flags
-without running every Solo command manually.
+Use One-shot Falcon deployment when you need a repeatable advanced setup, want to check a complete deployment into source control, or need to customise component flags without running every Solo command manually.
 
 Falcon is especially useful for:
 
@@ -36,19 +34,8 @@ Falcon is especially useful for:
 
 Before proceeding, ensure you have completed the following:
 
-- [**System Readiness**](/docs/simple-solo-setup/system-readiness) -your local environment
-  meets the hardware and software requirements for Solo, Kubernetes, Docker,
-  Kind, kubectl, and Helm.
-- [**Quickstart**](/docs/simple-solo-setup/quickstart) -you are already familiar with the
-  standard one-shot deployment workflow.
-- Set your environment variables if you have not already done so:
-
-  ```bash
-  export SOLO_CLUSTER_NAME=solo
-  export SOLO_NAMESPACE=solo
-  export SOLO_CLUSTER_SETUP_NAMESPACE=solo-cluster
-  export SOLO_DEPLOYMENT=solo-deployment
-  ```
+- [**System Readiness**](/docs/simple-solo-setup/system-readiness) - your local environment meets the hardware and software requirements for Solo, Kubernetes, Docker, Kind, kubectl, and Helm.
+- [**Quickstart**](/docs/simple-solo-setup/quickstart) -you are already familiar with the standard one-shot deployment workflow.
 
 ## How Falcon Works
 
@@ -64,8 +51,7 @@ sequence used by its one-shot workflows:
 7. Optionally, deploy mirror node, explorer, and relay in parallel for faster
    startup.
 8. Create predefined test accounts.
-9. Write deployment notes, versions, port-forward details, and account data to a
-   local output directory.
+9. Write deployment notes, versions, port-forward details, and account data to a local output directory.
 
 The difference is that Falcon reads a YAML file and maps its top-level sections
 to the underlying Solo subcommands.
@@ -131,14 +117,14 @@ solo one-shot falcon deploy --values-file falcon-values.yaml
 Solo creates a one-shot deployment, applies the values from the YAML file to the
 appropriate subcommands, and then deploys the full environment.
 
-### What Falcon Does Not Read from the File
+### Command-Line Flags (Not in YAML File)
 
-Some Falcon settings are controlled directly by the top-level command flags,
-not by section entries in the values file:
+The following flags are passed on the command line and cannot be set in the YAML file:
+- `--deployment`, `--namespace`, `--cluster-ref`, `--num-consensus-nodes`
+
+Note: `--values-file` specifies which YAML file to load.
 
 - `--values-file` selects the YAML file to load.
-- `--deploy-mirror-node`, `--deploy-explorer`, and `--deploy-relay` control
-  whether those optional components are deployed at all.
 - `--deployment`, `--namespace`, `--cluster-ref`, and `--num-consensus-nodes`
   are top-level one-shot inputs.
 
@@ -214,36 +200,6 @@ want to manage each step manually.
 > [**System Readiness**](/docs/simple-solo-setup/system-readiness), and increase Docker
 > memory and CPU allocation before deploying.
 
-## (Optional) Component Toggles
-
-Falcon can skip optional components at the command line without requiring a
-second YAML file.
-
-For example, to deploy only the consensus network and mirror node:
-
-```bash
-solo one-shot falcon deploy \
-  --values-file falcon-values.yaml \
-  --deploy-explorer=false \
-  --deploy-relay=false
-```
-
-Available toggles and their defaults:
-
-| Flag | Default | Description |
-| ---- | ------- | ----------- |
-| `--deploy-mirror-node` | `true` | Include the mirror node in the deployment. |
-| `--deploy-explorer` | `true` | Include the explorer in the deployment. |
-| `--deploy-relay` | `true` | Include the JSON RPC relay in the deployment. |
-
-> **Important**: The explorer and relay both depend on the mirror node. Setting `--deploy-mirror-node=false` while keeping `--deploy-explorer=true` or `--deploy-relay=true` is not a supported configuration and will result in a failed deployment.
-
-This is useful when you want to:
-
-- Reduce resource usage in CI jobs.
-- Isolate one component during testing.
-- Reuse the same YAML file across multiple deployment profiles.
-
 ## Common Falcon Customisations
 
 Because each YAML section maps directly to the corresponding Solo subcommand,
@@ -283,7 +239,7 @@ Falcon can also include block node configuration.
 > **Note:** Block node workflows are advanced and require higher resource
 > allocation and version compatibility across consensus node, block node, and
 > related components.
->Docker memory must be set to at least 16 GB before deploying with block node enabled.
+> Docker memory must be set to at least 16 GB before deploying with block node enabled.
 >
 > Block node support also requires the
 > `ONE_SHOT_WITH_BLOCK_NODE=true` environment variable to be set before
@@ -327,24 +283,6 @@ relayNode:
 Use block node settings only when your target Solo and component versions are
 known to be compatible.
 
-## Rollback and Failure Behaviour
-
-Falcon deployment enables automatic rollback by default.
-
-If deployment fails after resources have already been created, Solo attempts to
-destroy the one-shot deployment automatically and clean up the namespace.
-
-If you want to preserve the failed deployment for debugging, disable rollback:
-
-```bash
-solo one-shot falcon deploy \
-  --values-file falcon-values.yaml \
-  --no-rollback
-```
-
-Use `--no-rollback` only when you explicitly want to inspect partial resources,
-logs, or Kubernetes objects after a failed run.
-
 ## Deployment Output
 
 After a successful Falcon deployment, Solo writes deployment metadata to
@@ -363,18 +301,9 @@ This directory typically contains:
 This makes Falcon especially useful for automation, because the deployment
 artifacts are written to a predictable path after each run.
 
-To inspect the latest one-shot deployment metadata later, run:
+To inspect deployment output, check the `~/.solo/one-shot-<deployment>/` directory directly.
 
-```bash
-solo one-shot show deployment
-```
-
-If port-forwards are interrupted after deployment - for example after a system
-restart or network disruption - restore them without redeploying:
-
-```bash
-solo deployment refresh port-forwards
-```
+If port-forwards are interrupted after deployment, restore them by rerunning the component commands (such as `solo consensus node start`, `solo mirror node add`, etc.)
 
 ## Destroy a Falcon Deployment
 

--- a/content/en/docs/advanced-solo-setup/solo-ci-workflow.md
+++ b/content/en/docs/advanced-solo-setup/solo-ci-workflow.md
@@ -71,12 +71,14 @@ Install Kind to create and manage a local Kubernetes cluster in your workflow.
       uses: helm/kind-action@a1b0e391336a6ee6713a0583f8c6240d70863de3
       with:
         install_only: true
-        node_image: kindest/node:v1.31.4@sha256:2cb39f7295fe7eafee0842b1052a599a4fb0f8bcf3f83d96c7f4864c357c6c30
-        version: v0.26.0
-        kubectl_version: v1.31.4
+        node_image: kindest/node:v1.32.2@sha256:3966f21e12b760f6585bde7140cae5e8cdc0e52b37a6f90ce39834b6e72e3f49
+        version: v0.29.0
+        kubectl_version: v1.32.2
         verbosity: 3
         wait: 120s
   ```
+  > **Important:** Kind version must be **v0.29.0 or later** and  Kubernetes version must be **v1.32.2 or late**r. Solo enforces this 
+  > minimum version at runtime. Installing an older version will cause deployment failures.
 
 ## Step 2: Install Node.js
 
@@ -97,7 +99,7 @@ breaking changes from newer releases and cause unexpected workflow failures.
     - name: Install Solo CLI
       run: |
         set -euo pipefail
-        npm install -g @hiero-ledger/solo@0.68.0
+        npm install -g @hiero-ledger/solo@<version>
         solo --version
         kind --version
   ```
@@ -124,50 +126,108 @@ a fully functional local Hiero network, including:
         solo one-shot single deploy | tee solo-deploy.log
   ```
 
+## Cleanup
+
+After the workflow completes, destroy the Solo deployment and delete the Kind
+cluster to avoid leaving resources behind.
+
+  ```yaml
+    - name: Destroy Solo deployment
+      env:
+        SOLO_DEPLOYMENT: solo-deployment
+      run: |
+        set -euo pipefail
+        solo one-shot single destroy --deployment "${SOLO_DEPLOYMENT}"
+  ```
+
+  ```yaml
+    - name: Delete Kind cluster
+      env:
+        SOLO_CLUSTER_NAME: solo
+      run: |
+        set -euo pipefail
+        kind delete cluster -n "${SOLO_CLUSTER_NAME}"
+  ```
+
 ## Complete Example Workflow
 
 The following is the full workflow combining all steps above. Copy this into your
 .github/workflows/ directory as a starting point.
 
 ```yaml
+name: Solo CI Example
 
-  - name: Check Docker Resources
-    run: |
-      read cpus mem <<<"$(docker info --format '{{.NCPU}} {{.MemTotal}}')"
-      mem_gb=$(awk -v m="$mem" 'BEGIN{printf "%.1f", m/1000000000}')
-      echo "CPU cores: $cpus"
-      echo "Memory: ${mem_gb} GB"
-      
-  - name: Setup Kind
-    uses: helm/kind-action@a1b0e391336a6ee6713a0583f8c6240d70863de3
-    with:
-      install_only: true
-      node_image: kindest/node:v1.31.4@sha256:2cb39f7295fe7eafee0842b1052a599a4fb0f8bcf3f83d96c7f4864c357c6c30
-      version: v0.26.0
-      kubectl_version: v1.31.4
-      verbosity: 3
-      wait: 120s
+on:
+  workflow_dispatch:
+    inputs:
+      solo_version:
+        description: 'Solo CLI version to install'
+        required: false
+        default: '0.69.0'
+      kind_version:
+        description: 'Kind version to install (minimum v0.29.0)'
+        required: false
+        default: 'v0.29.0'
+      kubectl_version:
+        description: 'kubectl version to install (minimum v1.32.2)'
+        required: false
+        default: 'v1.32.2'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check Docker Resources
+        run: |
+          read cpus mem <<<"$(docker info --format '{{.NCPU}} {{.MemTotal}}')"
+          mem_gb=$(awk -v m="$mem" 'BEGIN{printf "%.1f", m/1000000000}')
+          echo "CPU cores: $cpus"
+          echo "Memory: ${mem_gb} GB"
+          
+      - name: Setup Kind
+        uses: helm/kind-action@a1b0e391336a6ee6713a0583f8c6240d70863de3
+        with:
+          install_only: true
+          node_image: kindest/node:v1.32.2@sha256:3966f21e12b760f6585bde7140cae5e8cdc0e52b37a6f90ce39834b6e72e3f49
+          version: v0.29.0
+          kubectl_version: v1.32.2
+          verbosity: 3
+          wait: 120s
          
-  - name: Set up Node.js
-    uses: actions/setup-node@49933ea5288caeca8642d1e84afbd3f7d6820020
-    with:
-      node-version: 22.12.0
+      - name: Set up Node.js
+        uses: actions/setup-node@49933ea5288caeca8642d1e84afbd3f7d6820020
+        with:
+          node-version: 22.12.0
       
-  - name: Install Solo CLI
-    run: |
-      set -euo pipefail
-      npm install -g @hiero-ledger/solo@0.68.0
-      solo --version
-      kind --version
+      - name: Install Solo CLI
+        run: |
+          set -euo pipefail
+          npm install -g @hiero-ledger/solo@<version>
+          solo --version
+          kind --version
       
-  - name: Deploy Solo
-    env:
-      SOLO_CLUSTER_NAME: solo
-      SOLO_NAMESPACE: solo
-      SOLO_CLUSTER_SETUP_NAMESPACE: solo-cluster
-      SOLO_DEPLOYMENT: solo-deployment
-    run: |
-      set -euo pipefail
-      kind create cluster -n "${SOLO_CLUSTER_NAME}"
-      solo one-shot single deploy | tee solo-deploy.log
+      - name: Deploy Solo
+        env:
+          SOLO_CLUSTER_NAME: solo
+          SOLO_NAMESPACE: solo
+          SOLO_CLUSTER_SETUP_NAMESPACE: solo-cluster
+          SOLO_DEPLOYMENT: solo-deployment
+        run: |
+          set -euo pipefail
+          kind create cluster -n "${SOLO_CLUSTER_NAME}"
+          solo one-shot single deploy | tee solo-deploy.log
+
+      - name: Destroy Solo deployment
+        env:
+          SOLO_DEPLOYMENT: solo-deployment
+        run: |
+          set -euo pipefail
+          solo one-shot single destroy --deployment "${SOLO_DEPLOYMENT}"
+
+      - name: Delete Kind cluster
+        env:
+          SOLO_CLUSTER_NAME: solo
+        run: |
+          set -euo pipefail
+          kind delete cluster -n "${SOLO_CLUSTER_NAME}"
 ```

